packages/auth/src/handlers.ts

```diff
@@ -128,24 +133,60 @@ export const requests = {
       throw new Error(`No user found with email: ${email}`)
     }
 
-    const groupUpdateData = {
-      schemas: ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
-      Operations: [
-        {
-          op: operation,
-          path: 'members',
-          value: [
+    if (operation === Wso2GroupOperation.Add) {
+      const hasSponsorEngagementTool =
+        Array.isArray(user.groups) &&
+        user.groups.some((group) => typeof group === 'object' && group.display === ODP_ROLE)
+
+      if (hasSponsorEngagementTool) {
+        return
+      }
+    }
+
+    let groupUpdateData: GroupUpdateData
+
+    switch (operation) {
+      case Wso2GroupOperation.Add:
+        groupUpdateData = {
+          schemas: ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
+          Operations: [
             {
-              display: user.userName, // The user's username
-              value: user.id, // The user's SCIM ID
+              op: Wso2GroupOperation.Add,
+              value: {
+                members: [
+                  {
+                    display: user.userName, // The user's username
+                    value: user.id, // The user's SCIM ID
+                  },
+                ],
+              },
             },
           ],
-        },
-      ],
-    }
+        }
+        break
 
-    const response = await api.patch(`/scim2/Groups/${group}`, groupUpdateData)
+      case Wso2GroupOperation.Remove:
+        groupUpdateData = {
+          schemas: ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
+          Operations: [
+            {
+              op: Wso2GroupOperation.Remove,
+              path: `members[value eq ${user.id}]`, // The user's SCIM ID
+            },
+          ],
+        }
+        break
+    }
 
-    return updateGroupResponseSchema.safeParse(response.data)
+    try {
+      const response = await requests.patchUserGroup(groupId, groupUpdateData)
+      return updateGroupResponseSchema.safeParse(response.data)
+    } catch (error) {
```

Comment:
I think axios defaults to throw an exception on anything other than 2xx.
Are we overriding validate status to change this behaviour anywhere?

https://axios-http.com/docs/handling_errors

Can we check in a running environment that this is catching the errors we want?

Reply:
yes, defaults to throw and exception, no overriding `validateStatus`
two exception we were seeing, one for users that already had the role, and one where users didn't have an IDG account.
already has the role - now covered by the eligibility check, we don't try to add when it already exists
no idg account - catching this and logging the error details, ran locally and catching a `500` and logging the response detail